client/src/components/Feed/PostSocial.js

```diff
@@ -42,71 +48,107 @@ const PostSocial = ({
     return liked ? (
       <SvgIcon src={heart} className="social-icon-svg" />
     ) : (
-      <SvgIcon src={heartGray} className="social-icon-svg" />
-    );
+        <SvgIcon src={heartGray} className="social-icon-svg" />
+      );
   };
 
   const renderCommentIcon = () => {
-    return showComments ? (
-      <SvgIcon src={commentGray} className="social-icon-svg" />
-    ) : (
-      <SvgIcon src={comment} className="social-icon-svg" />
-    );
+    return showComments ||
+      numComments > 0 ||
+      (history &&
+        history.location &&
+        history.location.state &&
+        history.location.state.comments) ? (
```

Comment:
Curious why the 3rd condition to display the blue heart icon. The first two make sense, but the 3rd using the history state, is hard to understand the rationale for. Maybe worth a comment? 
```
history &&
        history.location &&
        history.location.state &&
        history.location.state.comments
```

Also, FYI & optional, on client only we can use safe navigation operator for shorter code, e.g.: `history?.location?.state?.comments`

Reply:
I thought optional chaining was a Typescript thing. I didn't know you could use it in JavaScript. The last time I used JS was 2.5 years ago or so. This is super neat. Thanks!

Reply:
yeah it's ECMAScript and available in our current create-react-app version
but not on the backend with our Node.js version, or we would have to add babel transpilation